packages/zipkin-instrumentation-hapi/test/integrationTest.js

```diff
@@ -1,185 +1,195 @@
+          return reply.response(request.headers).code(200);
+        }
+      }
+    });
+    server.route({
+      method: 'GET',
+      path: '/weather/securedTown',
+      config: {
+        handler: (request, reply) => {
+          tracer.recordBinary('city', 'securedTown');
+          return reply.response(request.headers).code(401);
+        }
+      }
+    });
+    server.route({
+      method: 'GET',
+      path: '/weather/bagCity',
+      config: {
+        handler: () => {
+          tracer.recordBinary('city', 'bagCity');
+          throw new Error('service is dead');
+        }
+      }
+    });
+    server.route({
+      method: 'GET',
+      path: '/slow',
+      config: {
+        handler: (request, reply) => new Promise((resolve) => {
+          setTimeout(
+            () => resolve(reply.response(request.headers).code(202)),
+            PAUSE_TIME_MILLIS
+          );
         })
-        .then(() => {
-          const annotations = record.args.map(args => args[0]);
-          annotations.forEach(ann => expect(ann.traceId.traceId).to.equal('aaa'));
-          annotations.forEach(ann => expect(ann.traceId.spanId).to.equal('bbb'));
-
-          expect(annotations[0].annotation.annotationType).to.equal('ServiceName');
-          expect(annotations[0].annotation.serviceName).to.equal(serviceName);
-
-          expect(annotations[1].annotation.annotationType).to.equal('Rpc');
-          expect(annotations[1].annotation.name).to.equal('POST');
-
-          expect(annotations[2].annotation.annotationType).to.equal('BinaryAnnotation');
-          expect(annotations[2].annotation.key).to.equal('http.path');
-          expect(annotations[2].annotation.value).to.equal('/foo');
-
-          expect(annotations[3].annotation.annotationType).to.equal('ServerRecv');
+      }
+    });
+    server.register({
+      plugin: middleware,
+      options: {tracer}
+    }).then(() => server.start())
+      .then(() => {
+        baseURL = `http://127.0.0.1:${server.info.port}`;
+        done();
+      });
+  });
 
-          expect(annotations[4].annotation.annotationType).to.equal('LocalAddr');
+  afterEach(() => {
+    if (server) server.stop();
+    expect(spans).to.be.empty; // eslint-disable-line no-unused-expressions
+  });
 
-          expect(annotations[5].annotation.annotationType).to.equal('BinaryAnnotation');
-          expect(annotations[5].annotation.key).to.equal('http.status_code');
-          expect(annotations[5].annotation.value).to.equal('202');
+  function popSpan() {
+    expect(spans).to.not.be.empty; // eslint-disable-line no-unused-expressions
+    return spans.pop();
+  }
+
+  function successSpan(path, city) { // eslint-disable-line no-unused-vars
+    return ({
+      name: 'get',
+      kind: 'SERVER',
+      localEndpoint: {serviceName, ipv4},
+      tags: {
+        'http.path': path,
+        'http.status_code': '200',
+        // city TODO: scoping for the handler function
```

Comment:
ex test found that it is very likely downstream requests will break. if you cannot see the trace id to add a tag, it is likely you also cannot make a child span